plugins/single_plugins/grid.cpp

```diff
@@ -265,24 +288,29 @@ class wayfire_grid : public wf::plugin_interface_t
     void handle_slot(wayfire_view view, const wf_geometry workarea, int slot, bool comes_from_maximize_request)
     {
         wf_geometry target = get_slot_dimensions(slot, workarea);
-        if (view->maximized && view->get_wm_geometry() == target)
+        if (view->tiled_edges && view->get_wm_geometry() == target)
         {
-            return;
+            if (restore_opt_str == restore_opt_default)
+                slot = 0;
+            else
+                return;
         }
-        else if (!view->has_data<wf_grid_saved_view_geometry>() && slot)
+        if (!view->has_data<wf_grid_saved_view_geometry>() && slot)
         {
             view->get_data_safe<wf_grid_saved_view_geometry>()
                 ->geometry = view->get_wm_geometry();
         }
         else if (view->has_data<wf_grid_saved_view_geometry>() && slot == 0)
         {
-            target = view->get_data_safe<wf_grid_saved_view_geometry>()->geometry;
             /* If we have a request from another plugin, for ex. move, we want
              * to make sure the view stays under the pointer. However, when it was
              * a normal restore caused by a client request, we want to restore the
              * view to where it was */
-            if (!comes_from_maximize_request)
+            wf_geometry last_target = get_slot_dimensions(last_slot, workarea);
+            if (!comes_from_maximize_request && (view->get_wm_geometry() != last_target))
```

Comment:
can you explain why we need the second condition?

Reply:
I see it's wrong, will fix.